### DISCUSSION TITLE
Is there a way to determine mapbox performance?

### DISCUSSION THREAD

**Discussion by nonoumasy** [uber/react-map-gl#1943] [General]
For eg, I get conflicting reports on pages on my site. Some say they have issues, lag, etc while some don't see this issue on their end. Is there a way to measure performance programatically? On my end, I don't see any issues.

<img width="1440" alt="Screen Shot 2022-08-05 at 5 43 37 PM" src="https://user-images.githubusercontent.com/6312676/183226766-fee6a6cc-db90-4912-b738-d60bda4b2dbb.png">

Sample page:
https://history-maps.com/story/Greek-Persian-Wars

**Comment by ChristopherLyon** [comment]
An add, how hardware dependant is the library really? I also see a lot of variation in performance for even simple maps and initial load time. Do we have any best practices for a SWR type cache load then re-validate? Would save 500-600ms of looking at a blank screen for a lot of users, especially in globe view with fog.
